config/README.md

```diff
@@ -0,0 +1,83 @@
+# High Level Design
+
+This document is work in progress, some concepts are not yet available
+(e.g. Resolver is a private concept in the service for the moment).
+
+## Map
+
+The [Map](configmap.go) represents the raw configuration for a service (e.g. OpenTelemetry Collector).
+
+## Provider
+
+The [Provider](mapprovider.go) provides configuration, and allows to watch/monitor for changes.
```

Comment:
For concept which are types in the source code should we use the exact names we have in the source code, e.g.?
```suggestion
The [MapProvider](mapprovider.go) provides configuration, and allows to watch/monitor for changes.
```